scripts/email_manifest_tracking_letter_customer.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+import itertools
+from openpyxl import load_workbook
+import os
+
+from EPPs.common import SendMailEPP
+
+
+class EmailManifestLetter(SendMailEPP):
+    # populate the sample manifest with the sample date. Sample manifest template is determined by a step udf.
+    # The starting row and columns are determined by step UDFs.
+
+    _use_load_config = True  # should the config file be loaded?
+
+    # additional argument required to obtain the file location for newly create manifest in the LIMS step
+    def __init__(self, argv=None):
+        super().__init__(argv)
+        self.manifest = self.cmd_args.manifest
+        self.letter = self.cmd_args.letter
+
+    @staticmethod
+    def add_args(argparser):
+        argparser.add_argument(
+            '-m', '--manifest', type=str, required=True, help='Sample manifest generated by the LIMS',
+        )
+        argparser.add_argument(
+            '-t', '--letter', type=str, required=True, help='Sample tracking letter generated by the LIMS'
+        )
+
+    def _run(self):
+
+        if len(self.projects) > 1:
+            raise ValueError('More than one project present in step. Only one project per step permitted')
+
+
+        # obtain all of the inputs for the step
+        all_inputs = list(self.artifacts)
+        input_project_name=self.projects[0].name
+
+
+        # create list of unique containers
+        container_types = set()
+        unique_container_names=set()
+
+
+        for artifact in all_inputs:
+            container_types.add(artifact.container.type.name)
+            unique_container_names.add(artifact.container.name)
+
+
+        manifest_filepath=self.manifest + '-'+'Edinburgh_Genomics_Sample_Submission_Manifest_' + input_project_name + '.xlsx'
+        letter_filepath=self.letter+'-Edinburgh_Genomics_Sample_Tracking_Letter_'+all_inputs[0].samples[0].project.name+'.docx'
+
+
+        #send an email to the project manager using customer manifest template with the new manifest attached
+
+        #choose a species for the email subject, this can be updated manually by the project manager
+        species=all_inputs[0].samples[0].udf['Species']
```

Comment:
Wouldn't that give us something like `Homo sapiens, Homo sapiens, Homo sapiens, Homo sapiens, ...`? If that's the case, we;d need to pass it through `set`:
```python
species = ', '.join(set(s.udf['Species'] for s in self.samples))
```

Reply:
That is indeed correct!

Reply:
I did this intentionally because if there are multiple species the subject will become unreadable. However, this will be a rare event. It can be updated manually to give readability and it is probably better that the person releasing is reminded there are multiple species and the customer is not confused. Will update.